### PR TITLE
Rebuild with libcxx>=17

### PR DESCRIPTION
**Pull request by saraedum**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender

**Comment by saraedum** [comment]
@wlav is cppyy-cling going to work with clang 16 with root 63202 patches?

I am running in more problems because we have updated the libcxx in macOS and I guess that using a more recent clang colud help.

**Comment by saraedum** [comment]
So, essentially this https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4890 now breaks a lot of builds for me that depend on cppyy. On macOS, I am now forced to use libcxx>=17 but cppyy-cling forces libcxx<17 through the clangdev 13 package we are building against.

I think I'll have to disable macOS for the downstream packages of cppyy since it seems to be not usable here at the moment.

Any thoughts @isuruf?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -53,7 +54,7 @@ requirements:
     - pip
     - zlib
     # There is no root_62701 build of clangdev so we use the latest build before that
-    - clangdev =13.0.1 root_63002*
+    - clangdev =16 root_63202*
```
Comment (saraedum) [inline recipe/meta.yaml:57]:
```suggestion
    - clangdev =13 root_63004*
```

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender.

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender.

**Comment by wlav** [comment]
> @wlav is cppyy-cling going to work with clang 16 with root 63202 patches?
> 
> I am running in more problems because we have updated the libcxx in macOS and I guess that using a more recent clang colud help.

Upstream is at llvm16 (released), but is also very close to a release build on llvm18, so I was waiting on the latter. However, given these troubles, I'll start moving to llvm16 now.

**Comment by saraedum** [comment]
I managed to build this on macOS, I am not sure whether this is functional but maybe I am lucky.

**Comment by wlav** [comment]
I pushed the changed needed for LLVM16 into https://github.com/wlav/cppyy-backend/ . Still six failing tests on Linux. Haven't tried the other platforms yet.

**Pull request merged**